content/for-developers/tutorials/inner-loop/expose-metrics/expose-metrics.md

```diff
@@ -57,3 +57,28 @@ Now, let's add the `ServiceMonitor` for our `stakater-nordmart-review-api` appli
 Voila, you successfully exposed metrics for your application!
 
 Move on to the next tutorial to see how to trigger alerts for your application.
+
+## Troubleshooting
+
+If something doesn't work here are some commonly useful things to check:
+
+- Start by doing curl on localhost from within the pod with the metrics path to make sure that metrics are exposed.
+
+```sh
+oc get pods -n namespace
+oc debug pod/podname-from-above
+curl localhost:port/path
+```
+
+- Verify everything in spec in the ServiceMonitor.
+    - Verify that endpoints port matches with the Service spec.ports.name .
+    - Verify that namespaceSelector matchnames matches with the Service metadata.namespace .
+    - Verify that selector matchlabels matches with the Service metadata.labels .
+
+- Make sure that the service monitor selector and service monitor namespace selector labels have been applied to the service monitor and the namespace respectively where the service monitor is present. You can check these selectors in Search -> Resources -> Prometheus and then check the yaml manifest of Prometheus CR.
+
+- If nothing else works then start with port-forwarding the Prometheus pod to port: 9090 and navigate to Prometheus UI with http. Check in the service discovery that service monitor is picked up, also check the targets that Prometheus is able to scrape the metrics. If the target is visible, but metrics are not scraped then the error should be visible in targets.
```

Comment:
@Andtalath can we add some screenshots here to make it easy?

Reply:
Not really sure what screenshots are useful without also introducing confusing examples as well.

Reply:
We agreed outside of her that screenshots will more likely confuse users with too specific labels.